sr_ur_arm_calibration_loader/src/sr_ur_arm_calibration_loader/sr_ur_arm_calibration_loader.py

```diff
@@ -0,0 +1,161 @@
+            rospy.logerr('Cannot find default config for ' + CONST_ARM_TYPE)
+
+    def _setup_folders(self):
+        if not os.path.exists(self._arm_calibrations_folder):
+            os.makedirs(self._arm_calibrations_folder)
+
+    def _get_serial_from_arm(self, arm_ip):
+        try:
+            client = paramiko.SSHClient()
+            client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            client.connect(arm_ip, username=self._ur_arm_ssh_username, password=self._ur_arm_ssh_password)
+            stdin, stdout, stderr = client.exec_command('cat /root/ur-serial')
+            arm_serial_number = stdout.readline()
+            client.close()
+        except:
+            rospy.logerr("Error connceting to arm via SSH")
+            raise SSHError
+        if '' == arm_serial_number:
+            rospy.logwarn("Could not retrieve arm serial number via SSH, arm will NOT be calibrated.")
+            arm_serial_number = self._default_kinematics_config
+        return arm_serial_number
+
+    def _check_arm_calibration_exists(self, arm_serial):
+        arm_calibration_file = os.path.join(self._arm_calibrations_folder, arm_serial + '.yaml')
+        if os.path.isfile(arm_calibration_file):
+            return True
+        else:
+            return False
+
+    def _generate_new_arm_calibration(self, arm_ip, arm_serial):
+        try:
+            root = tk.Tk()
+            root.withdraw()
+            question_string = "No calibration detected for arm at " + arm_ip + ". Do you want to generate one?"
+        except:
+            rospy.logerr("Cannot create graphical prompt. If this is running over SSH, are SSH graphics enabled?")
+            raise SSHGraphicsError
+        answer = messageBox.askokcancel("Question", question_string)
+        root.destroy()
+        if answer is True:
+            self._start_calibration(arm_ip, arm_serial)
+            return True
+        else:
+            return False
+
+    def _get_yaml(self, filename):
+        with open(filename) as f:
+            data = yaml.load(f)
+        return data
+
+    def _start_calibration(self, arm_ip, arm_serial):
+        output_file = os.path.join(self._arm_calibrations_folder, arm_serial + ".yaml")
+        pkg = 'ur_calibration'
+        launch_file_name = 'calibration_correction.launch'
+        roslaunch_file = os.path.join(self._rospack.get_path(pkg), 'launch', launch_file_name)
+        robot_ip_arg = 'robot_ip:=' + arm_ip
+        output_filename_arg = 'target_filename:=' + output_file
+        cli_args = [roslaunch_file, output_filename_arg, robot_ip_arg]
+        uuid = roslaunch.rlutil.get_or_generate_uuid(None, False)
+        roslaunch.configure_logging(uuid)
+        roslaunch_args = cli_args[1:]
+        roslaunch_file = [(roslaunch.rlutil.resolve_launch_arguments(cli_args)[0], roslaunch_args)]
+        parent = roslaunch.parent.ROSLaunchParent(uuid, roslaunch_file)
+        parent.start()
+        rospy.sleep(8)
```

Comment:
why 8? can we somehow figure out exactly how much we need to wait?

Reply:
That's a timeout, it should return within that time